### PR TITLE
Switch High Contrast

### PR DESCRIPTION
**Pull request by nabil-k**
### Platforms Impacted
- [ ] iOS
- [ ] macOS
- [x] win32 (Office)
- [ ] windows
- [ ] android

### Description of changes

No longer depending on the values of the tokens in the redlines to provide the colors for high contrast. Instead now we are checking to see if we're in high contrast mode and map tokens we picked ourselves to provide the colors for high contrast.

This makes the Switch's high contrast look closer to what we expected a high contrast Switch to look like. It also resolves the issue of the thumb not being visible when using high contrast on windows 11.

### Verification

Windows 10 High Contrast
| Before                                       | After                                      |
|----------------------------------------------|--------------------------------------------|
| image | image |

Windows 11 Aquatic High Contract
Note: The first switch in these before/after screen shots has a mouse hovered over them.
| Before                                       | After                                      |
|----------------------------------------------|--------------------------------------------|
| image | image |

### Pull request checklist

This PR has considered (when applicable):
- [x] Automated Tests
- [ ] Documentation and examples
- [ ] Keyboard Accessibility
- [ ] Voiceover
- [ ] Internationalization and Right-to-left Layouts

### CODE REVIEW

File: packages/experimental/Switch/src/SwitchTokens.win32.ts [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+      thumbColor: isHighContrast(t) ? t.colors.compoundBrandBackground1Hover : t.colors.neutralForegroundInvertedLink,
+      borderColor: t.colors.compoundBrandBackground1Hover,
+    },
+    pressed: {
+      trackColor: t.colors.compoundBrandBackground1Pressed,
+      thumbColor: isHighContrast(t) ? t.colors.neutralForegroundOnBrandHoverPressed : t.colors.neutralForegroundInvertedLink,
+      borderColor: t.colors.compoundBrandBackground1Pressed,
+    },
+    disabled: {
+      trackColor: isHighContrast ? t.colors.neutralForegroundDisabled : t.colors.neutralBackgroundDisabled,
```
Comment (joannaquu) [inline packages/experimental/Switch/src/SwitchTokens.win32.ts:64]: Should this be isHighContrast(t) as well?

Comment (nabil-k) [reply packages/experimental/Switch/src/SwitchTokens.win32.ts]: thanks for catching that!

File: packages/experimental/Switch/src/SwitchTokens.win32.ts [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    toggleContainerFlexDirection: 'row',
+  },
+
+  after: {
+    flexDirection: 'row-reverse',
+    toggleContainerFlexDirection: 'row-reverse',
+  },
+
+  toggleOn: {
+    trackColor: t.colors.compoundBrandBackground1,
```
Comment (ksiler) [inline packages/experimental/Switch/src/SwitchTokens.win32.ts:49]: for tokens that don't check for `isHighContrast(t)`, are they already resolving to the right colors for high contrast?

**Review by rurikoaraki — APPROVED** [review]

File: packages/experimental/Switch/src/SwitchTokens.win32.ts (hunk 2 above)
Comment (nabil-k) [reply packages/experimental/Switch/src/SwitchTokens.win32.ts:49]: Yeah they are. When testing the control in high contrast all the colors seem to be appropriate.

**Review by ksiler — APPROVED** [review]

**Pull request merged**